File: example/script/appsync_client.ts
// ---------------------------------------------------------------
//	appsync_client.ts
//
//
// ---------------------------------------------------------------
// ts-node appsync_client.ts ../token.json
import axios from 'axios';
import fs from 'fs';

console.error('*** start ***');

const file_json = process.argv[2];

const json_str = fs.readFileSync(file_json, 'utf8');
const dict_aa = JSON.parse(json_str);
const access_token = dict_aa['AuthenticationResult']['IdToken'];

const url = 'https://graph-api.phi.blue/graphql';

const query = 'query {listQuests { items { Condition Value }}}';
const data = JSON.stringify({ query: query });

const headers = {
  headers: {
    'Content-Type': 'application/json',
    Authorization: access_token,
  },
};

axios
  .post(url, data, headers)
  .then(function (response: any) {
    console.log(JSON.stringify(response.data));
  })
  .catch(function (error: any) {
    console.log(error);
  })
  .then(function () {
    console.error('*** end ***');
  });

// ---------------------------------------------------------------

File: example/script/appsync_client.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	appsync_client.py
#
#
# ./appsync_client.py ../token.json
# ------------------------------------------------------------------
import  sys
import  json
import  requests
# ------------------------------------------------------------------
sys.stderr.write("*** start ***\n")

file_in = sys.argv[1]
with open(file_in) as ff:
	json_str = ff.read().strip()
dict_aa = json.loads(json_str)
access_token = dict_aa['AuthenticationResult']['IdToken']
#
url="https://graph-api.phi.blue/graphql"
#
headers = {'Authorization': '{}'.format(access_token)}
#
query='{ \"query\": \"query {listQuests { data { Condition Value }}}\" }'
response = requests.post(url, headers=headers, data=query)
#
print(response.text)
#
sys.stderr.write("*** end ***\n")
# ------------------------------------------------------------------
